main.cpp

```diff
@@ -95,11 +110,19 @@ struct FloatType
     FloatType& multiply ( float );
     FloatType& divide ( float );
 
-    operator float() { return *pointerToFloatValue; }
+    const FloatType& pow ( float );
+    const FloatType& pow ( double d_ ) { return pow(static_cast<float>(d_)); }
+    const FloatType& pow ( int i_ ) { return pow(static_cast<float>(i_)); }
```

Comment:
smart.  This hides the static_cast inside your class instead of forcing it to be done wherever you call `pow()` .  👍

Reply:
thanks